### ISSUE TITLE
boolean eq comparison too relaxed

### ISSUE THREAD

**Issue by haarg** [rurban/Cpanel-JSON-XS#117]
The current eq override used for booleans ends up forcing numeric comparisons.  This results in false values comparing equal to nearly any string.
```perl
Cpanel::JSON::XS::false eq "3"; # false
Cpanel::JSON::XS::false eq "true"; # false

Cpanel::JSON::XS::false eq ""; # true
Cpanel::JSON::XS::false eq "0.0"; # true
Cpanel::JSON::XS::false eq "True"; # true
Cpanel::JSON::XS::false eq "treu"; # true
Cpanel::JSON::XS::false eq "kfjdshsidugfhsdoafj"; # true
```

I can understand allowing false to compare equal to both "0" and "", although I do find it a bit questionable.  But comparing equal to any non-numeric string other than "true" is pretty obviously wrong.

**Comment by rurban** [comment]
Thanks, but duplicate of #116.

**Comment by haarg** [comment]
I'm not sure how this is a duplicate of #116.  The problems with the argument order will only potentially be an issue if you are comparing against another object.  This bug is about the actual comparisons being done later in the method.

**Comment by rurban** [comment]
Right. There's also a 3rd very critical endless recursion bug if both args are blessed into some JSON.

**Comment by rurban** [comment]
Fixed this and some more cornercases in 4.06. Thanks!
